code/modules/projectiles/guns/flamer.dm

```diff
@@ -117,37 +136,94 @@
 				if(do_after(user,magazine.reload_delay, TRUE, src, BUSY_ICON_GENERIC))
 					replace_magazine(user, magazine)
+					light_pilot(user,TRUE)
 				else
 					to_chat(user, "<span class='warning'>Your reload was interrupted!</span>")
 					return
 			else
 				replace_magazine(user, magazine)
+				light_pilot(user,TRUE)
 		else
 			current_mag = magazine
 			magazine.loc = src
 			replace_ammo(,magazine)
+			light_pilot(user,TRUE)
 
 	update_icon()
+
 	return TRUE
 
 /obj/item/weapon/gun/flamer/unload(mob/user, reload_override = 0, drop_override = 0)
 	if(!current_mag)
 		return FALSE //no magazine to unload
+	if(istype(current_mag, /obj/item/ammo_magazine/flamer_tank/backtank))
+		detach_fueltank(user)
+		return
 	if(drop_override || !user) //If we want to drop it on the ground or there's no user.
 		current_mag.forceMove(get_turf(src)) //Drop it on the ground.
 	else
 		user.put_in_hands(current_mag)
 
 	playsound(user, unload_sound, 25, 1)
+	light_pilot(user,FALSE)
 	user.visible_message("<span class='notice'>[user] unloads [current_mag] from [src].</span>",
 	"<span class='notice'>You unload [current_mag] from [src].</span>")
 	current_mag.update_icon()
 	current_mag = null
-
 	update_icon()
-
 	return TRUE
 
+/obj/item/weapon/gun/flamer/proc/attach_fueltank(mob/user, obj/item/ammo_magazine/flamer_tank/backtank/fueltank)
+	if (!fueltank || !istype(fueltank))
+		to_chat(user, "<span class='warning'>That's not an attachable fuel tank!</span>")
+		return
+	
+	if(fueltank.current_rounds <= 0)
+		to_chat(user, "<span class='warning'>That [fueltank.name] is empty!</span>")
+		return
+	
+	to_chat(user, "<span class='notice'>You begin linking [src] with the [fueltank.name]. Hold still...</span>")
+	if(do_after(user,fueltank.reload_delay, TRUE, src, BUSY_ICON_GENERIC))
+		if (current_mag)
+			if(istype(current_mag,/obj/item/ammo_magazine/flamer_tank/backtank))
+				detach_fueltank(user,FALSE)
+			else
+				user.put_in_hands(current_mag)//We remove the fuel tank if there is one
+		current_mag = fueltank
+		fueltank.attached_flamer = src
+		replace_ammo(user, fueltank)
+		light_pilot(user,TRUE)
+		playsound(user, reload_sound, 25, 1, 5)
+		update_icon(user)
+		var/obj/screen/ammo/A = user.hud_used.ammo
+		A.update_hud(user)
+	else
+		to_chat(user, "<span class='warning'>Your action was interrupted!</span>")
+		return
+
+/obj/item/weapon/gun/flamer/proc/detach_fueltank(mob/user,volontary = TRUE)
+	var/obj/item/ammo_magazine/flamer_tank/backtank/fueltank = current_mag
+	current_mag = null
+	fueltank?.attached_flamer=null
+	if (volontary)
+		to_chat(user, "<span class='notic'>You detach the fuel tank</span>")
+	playsound(user, unload_sound, 25, 1)
+	light_pilot(user,FALSE)
+	update_icon(user)
+	var/obj/screen/ammo/A = user.hud_used.ammo
+	A.update_hud(user)
+	
+	
+/obj/item/weapon/gun/flamer/dropped_new_loc(mob/user)
+	..()
+	var/mob/living/carbon/human/humanuser = user
+	if (istype(humanuser))
```

Comment:
when would it not be a human?

Reply:
A monkey?

Reply:
Monkeys shouldnt be using flamers anyway
They should also be a human species so just remove it